### DISCUSSION TITLE
FHIR Research study resource generation

### DISCUSSION THREAD

**Discussion by rohitkadam1410** [synthetichealth/synthea#1281] [Q&A]
Can we create reasearch study, research subject resource with the current version of synthea ?

**Comment by jawalonoski** [comment]
The FHIR exporters do not currently support those Resources.

You have two options:

1. Write your own code to post-process the Synthea data to add the FHIR resources for each Patient Bundle according to your use-case criteria.
2. Modify the Synthea code to generate these resources at runtime. There are a lot of ways you could do this, and none of them are straightforward. Maybe this is an area for us to write a detailed How-To article.

**Comment by rohitkadam1410** [comment]
Is there any reference which I can follow to generate this resource.

**Reply by jawalonoski** [reply]
"Reference" as in walk-through? No.

The code that exports FHIR is here... https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/export/FhirR4.java. You might find a clever method to generate a study resource given a certain event.

Alternatively, you create another resource in the module using the https://synthetichealth.github.io/module-builder/, perhaps a placeholder FHIR Procedure with a code like "XXXXX" that you can then find and replace that resource with the one you want through post-processing.

Alternatively, you could write your own https://github.com/synthetichealth/synthea/wiki/Health-Record-Editors that uses some custom criteria to identify patients and inserts the appropriate resource into their record.

There is no easy solution to your problem.
